### PR TITLE
impl debug namespace

### PR DESCRIPTION
**Pull request by jmcook1186**
Implements function in `debug` namespace.

I defined a new `Coordinate` struct because `ethereum_consensus::primitives:{Coordinate}` does not have the `execution_optimistic` field. I considered raising a PR against that repo to add the field but decided there might be knock-on effects of changes to that repo that are avoided by redefining the struct here. 

Related to #8

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -15,6 +15,14 @@ pub struct VersionData {
...
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Coordinate {
+    #[serde(with = "crate::serde::as_string")]
+    slot: Slot,
+    root: Root,
+    execution_optimistic: bool,
+}
+
```
Comment (ralexstokes) [inline src/types.rs]: if you wanted to do something like this: https://github.com/ralexstokes/beacon-api-client/blob/main/src/types.rs#L357

we could still reuse the `Coordinate` type from `ethereum-consensus` and flatten the other bits into a map

so you could try something like:

```rust
pub struct CoordinateWithMetadata {
    #[serde(flatten)]
    pub coordinate: Coordinate,
    #[serde(flatten)]
    pub meta: HashMap<String, serde_json::Value>,
}
```

it is minor here bc we only have the few fields but I do like transparently getting updates with a single definition vs many we have to manually synchronize

File: src/types.rs [hunk 2]
```diff
@@ -15,6 +15,14 @@ pub struct VersionData {
     pub version: String,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Coordinate {
+    #[serde(with = "crate::serde::as_string")]
+    slot: Slot,
+    root: Root,
+    execution_optimistic: bool,
```
Comment (ralexstokes) [inline src/types.rs]: if we keep this we will prolly want `pub` on all of these

File: src/api_client.rs [hunk 3]
```diff
@@ -4,26 +4,26 @@
...
-    RootData, StateId, SyncCommitteeDescriptor, SyncCommitteeDuty, SyncCommitteeSummary,
-    SyncStatus, ValidatorStatus, ValidatorSummary, Value, VersionData,
+    BlockId, CommitteeDescriptor, CommitteeFilter, CommitteeSummary, Coordinate, DepositContract,
+    EventTopic, FinalityCheckpoints, GenesisDetails, HealthStatus, PeerSummary, ProposerDuty,
+    PublicKeyOrIndex, RootData, StateId, SyncCommitteeDescriptor, SyncCommitteeDuty,
+    SyncCommitteeSummary, SyncStatus, ValidatorStatus, ValidatorSummary, Value, VersionData,
 };
 #[cfg(feature = "peer-id")]
 use crate::types::{NetworkIdentity, PeerDescription, PeerDescriptor};
+use ethereum_consensus::altair::mainnet::{
```
Comment (ralexstokes) [inline src/api_client.rs:13]: any reason these were re-imported for altair?

we will want some kind of solution to handle multiple phases/presets/configs but that would come later

**Review by ralexstokes — COMMENTED** [review]
nice work!

left some questions and looks pretty good

File: src/types.rs (hunk 1 above)
Comment (jmcook1186) [reply src/types.rs]: Ok, except there was an additional problem parsing the returned value directly into `slot` as type `Slot` for the imported `Coordinate`, hence the additional serde wrapper to wrangle it from `String` in the explicitly defined version.

File: src/types.rs (hunk 2 above)
Comment (jmcook1186) [reply src/types.rs]:
```suggestion
    pub slot: Slot,
    pub root: Root,
    pub execution_optimistic: bool,
```

File: src/api_client.rs (hunk 3 above)
Comment (jmcook1186) [reply src/api_client.rs:13]: ah, yea that can be reverted back to `phase0`. I originally thought the missing field might just be a versioning issue.
```suggestion
use ethereum_consensus::phase0::mainnet::{
```

File: src/types.rs (hunk 1 above)
Comment (ralexstokes) [reply src/types.rs]: ah, yeah you could add the `serde::as_string` bit to the `Slot` there in a PR if you want

I think its ok to duplicate this here if you want

Comment (ralexstokes) [reply src/types.rs]: can try this now w/ https://github.com/ralexstokes/ethereum-consensus/pull/161

will need to update the dependency rev in the `Cargo.toml` too

File: src/api_client.rs [hunk 4]
```diff
@@ -4,24 +4,26 @@
...
-    BlockId, CommitteeDescriptor, CommitteeFilter, CommitteeSummary, DepositContract, EventTopic,
-    FinalityCheckpoints, GenesisDetails, HealthStatus, PeerSummary, ProposerDuty, PublicKeyOrIndex,
-    RootData, StateId, SyncCommitteeDescriptor, SyncCommitteeDuty, SyncCommitteeSummary,
-    SyncStatus, ValidatorStatus, ValidatorSummary, Value, VersionData,
+    BlockId, CommitteeDescriptor, CommitteeFilter, CommitteeSummary, CoordinateWithMetadata,
+    DepositContract, EventTopic, FinalityCheckpoints, GenesisDetails, HealthStatus, PeerSummary,
+    ProposerDuty, PublicKeyOrIndex, RootData, StateId, SyncCommitteeDescriptor, SyncCommitteeDuty,
+    SyncCommitteeSummary, SyncStatus, ValidatorStatus, ValidatorSummary, Value, VersionData,
 };
+#[cfg(feature = "peer-id")]
```
Comment (ralexstokes) [inline src/api_client.rs]:
```suggestion
```

File: src/api_client.rs [hunk 5]
```diff
@@ -4,24 +4,26 @@
...
 use crate::types::{NetworkIdentity, PeerDescription, PeerDescriptor};
 use ethereum_consensus::altair::mainnet::{
     SignedContributionAndProof, SyncCommitteeContribution, SyncCommitteeMessage,
 };
 use ethereum_consensus::bellatrix::mainnet::{BlindedBeaconBlock, SignedBlindedBeaconBlock};
 use ethereum_consensus::builder::SignedValidatorRegistration;
+#[cfg(feature = "peer-id")]
```
Comment (ralexstokes) [inline src/api_client.rs]:
```suggestion
```

File: src/types.rs [hunk 6]
```diff
@@ -14,6 +14,14 @@ pub struct VersionData {
...
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct CoordinateWithMetadata {
+    #[serde(flatten)]
+    pub coordinate: Coordinate,
+    #[serde(flatten)]
+    pub meta: HashMap<String, serde_json::Value>,
+}
+
```
Comment (ralexstokes) [inline src/types.rs:24]: nice, I do have a slight preference for this

File: src/api_client.rs [hunk 7]
```diff
@@ -427,14 +434,17 @@ impl Client {
     }
 
     /* node namespace */
+    #[cfg(feature = "peer-id")]
```
Comment (ralexstokes) [inline src/api_client.rs]:
```suggestion
```

File: src/api_client.rs [hunk 8]
```diff
@@ -427,14 +434,17 @@ impl Client {
...
     pub async fn get_node_identity() -> Result<NetworkIdentity, Error> {
         unimplemented!("")
     }
 
+    #[cfg(feature = "peer-id")]
```
Comment (ralexstokes) [inline src/api_client.rs]:
```suggestion
```

File: src/api_client.rs [hunk 9]
```diff
@@ -427,14 +434,17 @@ impl Client {
...
     pub async fn get_node_peers(filters: &[PeerDescriptor]) -> Result<Vec<PeerDescription>, Error> {
         unimplemented!("")
     }
 
+    #[cfg(feature = "peer-id")]
```
Comment (ralexstokes) [inline src/api_client.rs]:
```suggestion
```

**Review by ralexstokes — APPROVED** [review]
nice updates!

**Pull request merged**